crates/router/src/connector/checkout.rs

```diff
@@ -78,14 +78,18 @@ impl ConnectorCommon for Checkout {
         res: types::Response,
     ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
         let response: checkout::ErrorResponse = if res.response.is_empty() {
+            let (error_codes, error_type) = if res.status_code == 401 || res.status_code == 422 {
```

Comment:
It makes sense to be returning an "invalid API key" error message for status code 401 (unauthorized), but I don't think returning the same error message for status code 422 (unprocessable content) is a good idea. Status code 422 is usually returned by a server when the request body could not be processed by the server. 

Please verify that this mapping is indeed intentional.